File: sample-rxjava/src/main/java/org/dbtools/sample/model/database/main/individual/IndividualManager.java
/*
 * IndividualManager.java
 *
 * Generated on: 05/07/2016 08:50:53
 *
 */



package org.dbtools.sample.model.database.main.individual;

import org.dbtools.query.sql.SQLQueryBuilder;
import org.dbtools.sample.model.database.DatabaseManager;


public class IndividualManager extends IndividualBaseManager {


    public IndividualManager(DatabaseManager databaseManager) {
        super(databaseManager);
    }


    public long findLastIndividualId() {
        return findValueByRawQuery(Long.class, "SELECT MAX(" + IndividualConst.C_ID + ") FROM " + IndividualConst.TABLE, null, 0L);
    }

    public Individual findLastIndividual() {
        return findByRowId(findLastIndividualId());
    }

    public String findFirstNameById(long id) {
        return findValueBySelection(String.class, IndividualConst.C_FIRST_NAME, IndividualConst.C_ID + " = ?", SQLQueryBuilder.toSelectionArgs(id), "");
    }
}

File: sample-basic-kotlin/src/main/kotlin/org/dbtools/sample/kotlin/model/database/main/individual/IndividualManager.kt
/*
 * IndividualManager.kt
 *
 * Generated on: 05/09/2016 09:21:13
 *
 */



package org.dbtools.sample.kotlin.model.database.main.individual

import org.dbtools.query.sql.SQLQueryBuilder
import org.dbtools.sample.kotlin.model.database.DatabaseManager


class IndividualManager(databaseManager: DatabaseManager) : IndividualBaseManager(databaseManager) {

    fun findLastIndividualId(): Long {
        return findValueByRawQuery(valueType = Long::class.java,
                rawQuery = "SELECT MAX(${IndividualConst.C_ID}) FROM ${IndividualConst.TABLE}",
                defaultValue = 0L)
    }

    fun findLastIndividual(): Individual? {
        return findByRowId(findLastIndividualId())
    }

    fun findFirstNameById(id: Long): String {
        return findValueBySelection(String::class.java, IndividualConst.C_FIRST_NAME, "", IndividualConst.C_ID + " = ?", SQLQueryBuilder.toSelectionArgs(id), "")
    }
}